### PR TITLE
feat: add stop() method to gracefully terminate unleash

### PR DESCRIPTION
**Pull request by dezitter**
Closes https://github.com/Unleash/unleash/issues/570

### CODE REVIEW

File: lib/db/client-instance-store.js [hunk 1]
```diff
@@ -133,6 +133,10 @@ class ClientInstanceStore {
             .where('app_name', appName)
             .del();
     }
+
+    destroy() {
+        clearInterval(this.timer);
```
Comment (ivarconr) [inline lib/db/client-instance-store.js:138]: :+1:

**Comment by ivarconr** [comment]
Let me know when you feel this is complete enough for me to look in to! :fire:

**Comment by dezitter** [comment]
You can have a look now 👍

**Comment by ivarconr** [comment]
So all internal timers are stopped and thats nice.

But Unleash express app is still attached the port and still accepts connections (and new db connection is automatically created when new requests arrives. 

Do you now a simple way to detach the express app from the port as well?

**Comment by ivarconr** [comment]
I found one thin we need to improve in the implementation;

there is no way to `stop` unleash when using the `create()` method ([disable auto-start](https://unleash.github.io/docs/getting_started.html#disabling-auto-start)):

```js
const unleash = require('unleash-server');
const app = express();

const instance = await unleash.create({
    databaseUrl: 'postgres://unleash_user:password@localhost:5432/unleash',
    port: 4242,
  });

instance.stop() <<-- not available
```

**Comment by dezitter** [comment]
That's a good point. I've added a similar `stop` method to handle this second use case.

File: lib/server-impl.test.js [hunk 2]
```diff
@@ -99,3 +107,13 @@ test('should not create a server using create()', async t => {
     });
     t.true(typeof server === 'undefined');
 });
+
+test.only('should shutdown the server when calling stop()', async t => {
```
Comment (0x203) [inline lib/server-impl.test.js:111]: This `.only` looks like an artefakt from local testing that shouldn't be merged, or should it?

Comment (dezitter) [reply lib/server-impl.test.js:111]: You are right sorry about that. I opened another PR to fix that https://github.com/Unleash/unleash/pull/674

Thanks !

**Pull request merged**